src/app/instances/instances.controller.js

```diff
@@ -0,0 +1,190 @@
+    	    		if( instance.path === $scope.selectedInstance.path  ) {
+    	    			if( instance.status !== $scope.selectedInstance.status ) {
+    	    				$scope.setSelectedInstance( instance );
+    	    			}
+    	    			
+    	    			break;
+    	    		}
+    	    	}
+        	}
+        };
+        
+        $scope.isParentNotDeployed = function( instance ) {
+        	
+        	var node = instance.treeNode;
+        	while( node.parent ) {
+        		node = node.parent;
+        	}
+        	
+        	return node.instance.status === 'NOT_DEPLOYED';
+        };
+        
+        $scope.findTemplateUrl = function( status, isRoot, isParentNotDeployed ) {
+        	var result = '';
+        	
+        	if( status === 'NOT_DEPLOYED' ) {
+        		result = isRoot ? 'app-root-not-deployed.html' : isParentNotDeployed ? 'app-not-deployed-no-action.html' : 'app-not-deployed.html';
+        	 } else if( status === 'STARTING' ) {
+        		result = 'app-starting.html';
+        	 } else if( status === 'DEPLOYING' ) {
+        		result = isRoot ? 'app-root-deploying.html' : 'app-deploying.html';
+        	 } else if( status === 'UNDEPLOYING' ) {
+        		result = isRoot ? 'app-root-undeploying.html' : 'app-undeploying.html';
+        	 } else if( status === 'STOPPING' ) {
+        		result = 'app-stopping.html';
+        	 } else if( status === 'DEPLOYED_STOPPED' ) {
+        		result = 'app-deployed-stopped.html';
+        	 } else if( status === 'PROBLEM' ) {
+        		result = 'app-problem.html';
+        	 } else if( status === 'UNRESOLVED' ) {
+         		result = 'app-unresolved.html';
+        	 } else if( status === 'DEPLOYED_STARTED' ) {
+        		result = isRoot ? 'app-root-deployed-started.html' : 'app-deployed-started.html';
+        	 }
+        	
+        	return 'templates/instances/includes/' + result;
+        };
+        
+        $scope.findBlockClass = function( instancePath ) {
+        	var result = 'block short-block';
+        	if( $scope.selectedInstance ) {
+        		if( $scope.selectedInstance.path !== instancePath && 
+        				! $scope.selectedInstance.path.startsWith( instancePath + '/' )) {
+            		result += ' block-faded';
+        		}
+        	}
+        	
+        	return result;
+        };
+        
+        $scope.loadInstances = function() {
+        	Restangular.all( 'app/' + $scope.appName + '/children?all-children=true' ).getList().then( function( instances ) {
+        		$scope.rErrorMsg = '';
+        		$scope.rootNodes = rUtils.sortInstances( instances );
+        		$scope.invoked = true;
+        		
+        	}, function() {
+        		$scope.noError = false;
+        	});
+        };
+        
+        
+        // Initialize the page
+        $scope.loadInstances();
+        //$scope.updateFromServer();
+        
+        function initializeSearch() {
+        	
+        	// Show the search elements
+        	$( '#Finder' ).parent().show();
```

Comment:
Nope, never ever touch the DOM inside a controller. Use directives for this.

It's commonly admitted that capital letters in DOM ids & classes is bad practice (because browser side it's case insensitive).